flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/FlinkRelOptUtil.scala

```diff
@@ -495,4 +497,48 @@ object FlinkRelOptUtil {
     }
   }
 
+  /**
+   * Creates a projection which casts a rel's output to a desired row type.
+   *
+   * <p>This method is inspired by [[RelOptUtil.createCastRel]], different with that,
+   * we do not generate another Project if the rel is already a [[Project]].
+   *
+   * @param rel Producer of rows to be converted
+   * @param castRowType Row type after cast
+   * @return Conversion rel with castRowType
+   */
+  def createCastRel(
```

Comment:
TBH, I think it would be better to have this written in Java. We should try to add new code only in Java.

Reply:
Agree, would create a new Java class named RelOptUtils then put this method there.